### ISSUE TITLE
Mechanism to indicate a tasks is only executed for side-effects to prevent re-execution upon worker restart

### ISSUE THREAD

**Issue by benjeffery** [dask/dask#10654]
I have a function `parse_chunk` which writes it's output to disk and returns `None`. Running this function in the classic dask way:

```python
tasks = []
for c in chunks:
  tasks.append(dask.delayed(parse_chunk)(c))
dask.compute(*tasks)
```
On the dashboard I see the progress bar move forward as chunks are processed, but when a worker is rotated out (they have to be due to cluster time limits) the progress bar skips backward as all the results of the tasks done by that worker are lost. The results are not needed as the function just returns `None`.

One solution I have tried is this:

```python
futures = []
for c in chunks:
    futures.append(client.submit(parse_chunk,c))
for future in distributed.as_completed(futures):
        future.result()
        client.cancel(future)
```
Which works  - the work is no longer redone. But now the progress bar is very confusing for users as it stays at zero, but the work left to be done slowly decreases to zero.

Could there be a way to indicate that the results of a task are not needed and don't need to be held in cluster memory to the end of the `compute` call?

Many thanks!

**Comment by mrocklin** [comment]
I would do this:

```python
from dask.distributed import fire_and_forget

for x in L:
    future = client.submit(func, x)
    fire_and_forget(future)
```

https://docs.dask.org/en/latest/futures.html#fire-and-forget

**Comment by benjeffery** [comment]
I've tried this out, but I still need to wait for the set of futures to complete before execution moves on, If I do this with `distributed.wait` the futures that are executed by workers that have died are still being re-done. I've tried manually checking the futures, but keeping them in scope still forces re-execution.

**Comment by mrocklin** [comment]
Yeah, if you still have a future live in scope and it errors in some way or
goes away (such as due to a worker disappearing unexpectedly).  Sorry about
that.  I would probably stick with your as_completed solution and find some
other way to show progress maybe?

**Comment by mrocklin** [comment]
> but when a worker is rotated out (they have to be due to cluster time limits) the progress bar skips backward as all the results of the tasks done by that worker are lost

Another option is to think through this.  Ideally if a worker is closed gracefully its results (just the Nones) will be transferred to another worker.  If the worker is killed ungracefully then this doesn't happen.

Maybe you want to start your workers with a `--lifetime "1 hour"` keyword or something (or maybe `--lifetime "55 minutes"` would be better so that they kill themselves before the job manager kills them after an hour, say.

**Comment by benjeffery** [comment]
> Maybe you want to start your workers with a `--lifetime "1 hour"` keyword or something (or maybe `--lifetime "55 minutes"` would be better so that they kill themselves before the job manager kills them after an hour, say.

Sadly I can't always know when a worker will be killed. I've also thought about using the AMM to replicate the results - I'd need to be able add a policy (and remove the default `ReduceReplicas` policy) in library code to do this, as I'm working on a library that has other users so can't rely on dask config to configure the AMM. Are there docs for adding/removing policies on the fly in client code? Thanks for your help here.
